lib/matplotlib/tests/test_style.py

```diff
@@ -23,21 +25,16 @@ def temp_style(style_name, settings=None):
     if not settings:
         settings = DUMMY_SETTINGS
     temp_file = '%s.%s' % (style_name, STYLE_EXTENSION)
-
-    # Write style settings to file in the temp directory.
-    tempdir = tempfile.mkdtemp()
-    with open(os.path.join(tempdir, temp_file), 'w') as f:
-        for k, v in settings.items():
-            f.write('%s: %s' % (k, v))
-
-    # Add temp directory to style path and reload so we can access this style.
-    USER_LIBRARY_PATHS.append(tempdir)
-    style.reload_library()
-
     try:
-        yield
+        with TemporaryDirectory() as tmpdir:
+            # Write style settings to file in the tmpdir.
+            Path(tmpdir, temp_file).write_text(
+                "\n".join(map("{0[0]}: {0[1]}".format, settings.items())))
```

Comment:
I'd rather go with `"\n".join("{}: {}".format(k, v) for k, v in settings.items())`, which is more readable than mapping and tuple-item access in format.

Reply:
sure